docs/install.rst

```diff
@@ -257,32 +257,60 @@ Virtual environment
 Another method of installing Streamlink in a non-system-wide way is
 using `virtualenv`_, which creates a user owned Python environment instead.
 
-.. code-block:: bash
+- Install with ``virtualenv`` and ``pip`` commands
+
+  .. code-block:: bash
+
+      # Create a new environment
+      virtualenv ~/myenv
+
+      # Activate the environment
+      source ~/myenv/bin/activate
+
+      # *Either* install the latest Streamlink release from PyPI in the virtual environment
+      pip install --upgrade streamlink
+
+      # *Or*, install the most up-to-date development version from master on GitHub
+      pip install --upgrade git+https://github.com/streamlink/streamlink.git
+
+      # Use Streamlink in the environment
+      streamlink ...
+
+      # Deactivate the environment
+      deactivate
+
+      # Use Streamlink without activating the environment
+      ~/myenv/bin/streamlink ...
+
+- Install with ``pipx`` command
```

Comment:
If you want to add a `pipx` section, then it'd probably make more sense to add another subsection instead of splitting this up into a list with two items. `pipx` is just a wrapper for `virtualenv` and `pip`, so having a subsection makes more sense..

Reply:
I wasn't quite sure how to organise it.  Ideally, each would have been a sub section beneath the "Virtual environment" sub section, but I don't think that's possible?  So, maybe "Virtual environment with `pip`" and "Virtual environment with `pipx`" sub sections of "PyPI package and source code" sub section?

Reply:
Don't know what's the best. Let's have a look.